packages/pgen/cst/generator.ts

```diff
@@ -25,6 +25,7 @@ export const gen = async (source: string, target: string) => {
 }
 
 const PARSER_HEADER = `/* Generated. Do not edit. */
+// @ts-nocheck
```

Comment:
what problem does this solve? and why is it not at the top of the file?

Reply:
This solves the problem that TS doesn't understand `arr.length != 0 && arr[0]` smartcast, so we just disable checking for that file :)

Reply:
> why is it not at the top of the file?

because the first line is reserved for the "generated" mark

Reply:
fixed :) We don't need it now